arkouda/dataframe.py

```diff
@@ -4402,6 +4406,480 @@ def merge(
+        +----+------+-------+-----+-----+-----+
+        |  2 | True | False |   3 | c   |   3 |
+        +----+------+-------+-----+-----+-----+
+        |  3 | True | False |   4 | d   |   4 |
+        +----+------+-------+-----+-----+-----+
+
+        >>> df.dropna(axis=1, thresh=3)
+
+        +----+------+-----+-------+-----+-----+-----+
+        |    | A    |   C | D     |   E | F   |   G |
+        +====+======+=====+=======+=====+=====+=====+
+        |  0 | True |   1 | False |   1 | a   |   1 |
+        +----+------+-----+-------+-----+-----+-----+
+        |  1 | True |   2 | False |   2 | b   |   2 |
+        +----+------+-----+-------+-----+-----+-----+
+        |  2 | True |   3 | False |   3 | c   |   3 |
+        +----+------+-----+-------+-----+-----+-----+
+        |  3 | True | nan | False |   4 | d   |   4 |
+        +----+------+-----+-------+-----+-----+-----+
+
+        >>> df.dropna(axis=1, how="all")
+
+        +----+------+-----+-----+-------+-----+-----+-----+
+        |    | A    |   B |   C | D     |   E | F   |   G |
+        +====+======+=====+=====+=======+=====+=====+=====+
+        |  0 | True |   1 |   1 | False |   1 | a   |   1 |
+        +----+------+-----+-----+-------+-----+-----+-----+
+        |  1 | True | nan |   2 | False |   2 | b   |   2 |
+        +----+------+-----+-----+-------+-----+-----+-----+
+        |  2 | True |   2 |   3 | False |   3 | c   |   3 |
+        +----+------+-----+-----+-------+-----+-----+-----+
+        |  3 | True | nan | nan | False |   4 | d   |   4 |
+        +----+------+-----+-----+-------+-----+-----+-----+
+
+        """
+        from arkouda import all as akall
+
+        if (how is not None) and (thresh is not None):
+            raise TypeError("You cannot set both the how and thresh arguments at the same time.")
+
+        if how is None:
+            how = "any"
+
+        if (isinstance(axis, int) and axis == 0) or (isinstance(axis, str) and axis == "index"):
+            agg_axis = 1
+
+        elif (isinstance(axis, int) and axis == 1) or (isinstance(axis, str) and axis == "columns"):
+            agg_axis = 0
+
+        if thresh is not None:
+            counts = self.count(axis=agg_axis)
+            mask = counts >= thresh  # type: ignore
+        elif how == "any":
+            mask = self.notna().all(axis=agg_axis)
+        elif how == "all":
+            mask = self.notna().any(axis=agg_axis)
+        else:
+            raise ValueError(f"invalid how option: {how}")
+
+        if (isinstance(mask, bool) and mask is True) or (
+            isinstance(mask, Series) and akall(mask.values) is True
+        ):
+            result = self.copy(deep=None)
+        else:
+            if (isinstance(axis, int) and axis == 0) or (isinstance(axis, str) and axis == "index"):
+                if self.empty is True:
+                    result = DataFrame()
+                else:
+                    result = self[mask].copy(deep=True)
```

Comment:
I noticed that we have both deep and shallow copies in this one. Is there a reason to not be consistent in how we're copying?

also at some point do we want to try and add an `inplace` flag like in the pandas version? I haven't thought through it, so I'm mostly asking if you ran into problems adding it or if it was left out to get this functionality added sooner

Reply:
Admittedly that could be more consistent.  I was going to look into it further when I added the `inplace` flag, which I thought would be easier once Brandon's indexing updates were merged in.  I've already made tickets for the ways we could better align with pandas:
https://github.com/Bears-R-Us/arkouda/issues/3104
https://github.com/Bears-R-Us/arkouda/issues/3105
https://github.com/Bears-R-Us/arkouda/issues/3106
https://github.com/Bears-R-Us/arkouda/issues/3107

Reply:
okay awesome!